articles/build-an-outlook-clone-using-react-hooks/index.md

```diff
@@ -0,0 +1,881 @@
+
+React class-based components have a bad outlook towards code quality. There is a lot of repetition. DRY, an acronym that stands for Don't Repeat Yourself, effectively speaks for itself. Therefore, React Hooks is a layer on top of the existing library. Let's understand what hooks are and how they work in the next section.
+
+### React Hooks: Why?
+
+In the earlier article, we were constrained to use state and other React features inside the class. We defined the class as
+
+```jsx
+import React from 'react';
+
+class Example extends React.Component{
+    //sample component code here
+}
+```
+The above definition declares that *Example* class is a child component of *React.Component* class.
+
+### The basis of the need for react hooks in terms of the React Documentation page is as follows.
+
+#### *It’s hard to reuse stateful logic between components*
+
+The previous article had a lot of messy lines, where all the info that had to be sent to the child components were sent via the props. What if cases arise, where information has to be sent multiple levels downwards? In class-based components, it becomes messy due to layers of providers, consumers, render props, and higher-order components. Overall this points to an underlying problem of inability to sharing stateful info.
+
+
+With Hooks, you can extract stateful logic from a component so it can be tested independently and reused. Hooks allow you to reuse stateful logic without changing your component hierarchy.
+
+#### *Complex components become hard to understand*
+
+Usually projects that start out small, in no time, turn out to be an unmanageable mess of stateful logic and side effects. The order of the lifecycle methods is important for absolute control over the web page. However, consider the example where components might perform some data fetching in componentDidMount and componentDidUpdate. At the same time, another piece of code is present in componentDidUpdate setting up event listeners. The event listeners need to be set free, and this is done upon componentWillUnmount. So many lifecycle methods, with dependency on multiple non-related pieces of code, are problematic. Usually, state management libraries like redux are used to manage the state. However, with the introduction of Hooks, the above problem is solved.
+
+#### *Classes are confusing, Functions enable modularity*
+
+We used a lot of the *this* keyword in JavaScript in the earlier article. For beginners, learning *this* in JavaScript is quite a steep learning curve. Additionally, the React team found that classes inhibited higher performance.  
```

Comment:
> Additionally, the React team found that classes inhibited higher performance.

 If you are going to make statements like this you need to provide a link to your source.